tools/Elasticurl/Elasticurl.cs

```diff
@@ -0,0 +1,426 @@
+                        case "-i":
+                            ctx.IncludeHeaders = true;
+                            break;
+                        case "--insecure":
+                        case "-k":
+                            ctx.Insecure = true;
+                            break;
+                        case "--method":
+                        case "-M":
+                            ctx.Verb = NextArg(args, ref argIdx);
+                            break;
+                        case "--output":
+                        case "-o":
+                            ctx.OutputFilename = NextArg(args, ref argIdx);
+                            break;
+                        case "--post":
+                        case "-P":
+                            ctx.Verb = "POST";
+                            break;
+                        case "--trace":
+                        case "-t":
+                            ctx.TraceFile = NextArg(args, ref argIdx);
+                            break;
+                        case "--verbose":
+                        case "-v":
+                            string level = NextArg(args, ref argIdx);
+                            bool found = false;
+                            foreach (LogLevel value in Enum.GetValues(typeof(LogLevel)))
+                            {
+                                if (string.Equals(level, value.ToString(), StringComparison.OrdinalIgnoreCase))
+                                {
+                                    ctx.LogLevel = value;
+                                    found = true;
+                                    break;
+                                }
+                            }
+                            if (!found)
+                            {
+                                Console.WriteLine("Invalid log level: {0}", level);
+                                ShowHelp();
+                                Environment.Exit(-1);
+                            }
+                            break;
+                        default:
+                            ShowHelp();
+                            Environment.Exit(0);
+                            return;
+                    }
+                }
+
+                ctx.Uri = new Uri(args[args.Length - 1]);
+            }
+            catch (IndexOutOfRangeException)
+            {
+                ShowHelp();
+                Environment.Exit(-1);
+            }
+            catch (UriFormatException ufe)
+            {
+                Console.WriteLine("Invalid URI: {0}: {1}", args[args.Length - 1], ufe.Message);
+                Environment.Exit(-1);
+            }
+        }
+
+        static void InitLogging()
+        {
+            Logger.EnableLogging(ctx.LogLevel, ctx.TraceFile);
+        }
+
+        static void InitOutput()
+        {
+            if (ctx.OutputFilename != null)
+            {
+                try {
+                    ctx.OutputStream = File.OpenWrite(ctx.OutputFilename);

```

Comment:
do you need the binary flag on this one?

Reply:
FileStream is binary